src/libasr/diagnostics.cpp

```diff
@@ -62,17 +62,22 @@ bool Diagnostics::has_error() const {
 }
 
 std::string Diagnostics::render(const std::string &input,
-        const LocationManager &lm, const CompilerOptions &compiler_options) {
+        LocationManager &lm, const CompilerOptions &compiler_options) {
     std::string out;
     for (auto &d : this->diagnostics) {
         if (compiler_options.no_warnings && d.level != Level::Error) {
             continue;
         }
         if (compiler_options.error_format == "human") {
-            out += render_diagnostic_human(d, input, lm,
+            std::string input1;
+            LFortran::read_file(lm.in_filename.back(), input1);
+            lm.init_simple(input1);
+std::cout << lm.in_filename.back()+"\n"+input1;
+            out += render_diagnostic_human(d, input1, lm,
                 compiler_options.use_colors,
                 compiler_options.show_stacktrace);
             if (&d != &this->diagnostics.back()) out += "\n";
+            lm.in_filename.pop_back();
```

Comment:
I'm sure, whether these changes can be made.

Reply:
Can you show the output with these changes?